src/cfront.c

```diff
@@ -1308,6 +1309,142 @@ int read_body_assignment(char *token, block_t *parent)
     return 0;
 }
 
+int read_numeric_sconstant()
+{
+    /* return signed constant */
+    int isneg = 0, res;
+    char buffer[10];
+    if (lex_accept(T_minus))
+        isneg = 1;
+    if (lex_peek(T_numeric, buffer))
+        res = read_numeric_constant(buffer);
+    else
+        error("Invalid value after assignment");
+    lex_expect(T_numeric);
+    if (isneg)
+        return -1 * res;
+    else
+        return res;
+}
+
+int get_expression_imm(opcode_t op, int op1, int op2)
```

Comment:
The verb `get` might not reflect what the function does. It can be `evaluate` or `eval` in short.